l1-contracts/src/core/Decoder.sol

```diff
@@ -236,6 +250,15 @@ contract Decoder {
       offsets.contractDataOffset = offsets.contractOffset + lengths.contractCount * 0x20;
       offsets.l1ToL2MessagesOffset = offsets.contractDataOffset + 0x4 + lengths.contractCount * 0x34;
 
+      // load the l2 to l1 msgs
+      assembly {
```

Comment:
Can this array copy be moved down to where the other one is performed?

Reply:
I'm using that we just computer the offset values. We will be updating them every loop so was a nice way to have it right away.

Reply:
understood